class-instant-articles-post.php

```diff
@@ -737,12 +737,25 @@ public function add_ads_from_settings() {
 			$height = intval( $dimensions_match[2] );
 		}
 
+		$source_of_ad = isset( $settings_ads['ad_source'] ) ? $settings_ads['ad_source'] : 'none';
+
+		$ad = $this->get_ads_for_ad_source($source_of_ad, $width, $height);
+
+		// Hook for Ads
+		if( has_filter('instant_articles_loading_ads') ){
+			$ad = apply_filters('instant_articles_loading_ads', $ad);
+		}
+		$header->addAd( $ad );
+
+		$this->instant_article->enableAutomaticAdPlacement();
+	}
+
+	public function get_ads_for_ad_source($ads_source, $width, $height) {
```

Comment:
You call this $ads_source but the function uses $source_of_ads, this would break everything.